tests/taker_cli.rs

```diff
@@ -0,0 +1,321 @@
+        rpc_network: String,
+        network: String,
+        data_dir: Option<PathBuf>,
+        taker: Taker, 
+        makers: Vec<Arc<Maker>>,
+        bitcoind: BitcoinD,
+    }
+
+    impl TakerCli {
+        fn new(
+            data_dir: Option<PathBuf>,
+            rpc_network: &str,
+            network: &str,
+            makers_count: u16,
+            wallet_name: Option<String>,
+        ) -> TakerCli {
+            //---------------------- start bitcoind--------------------------------------------------------
+            // Initiate bitcoind instance
+
+            let temp_dir = tempdir().unwrap().into_path();
+
+            let mut conf = Conf::default();
+
+            conf.args.push("-txindex=1"); //txindex is must, or else wallet sync won't work.
+            conf.staticdir = Some(temp_dir.join(".bitcoin"));
+
+            let key = "BITCOIND_EXE";
+            let curr_dir_path = std::env::current_dir().unwrap();
+            let bitcoind_path = curr_dir_path.join("bin").join("bitcoind");
+            std::env::set_var(key, bitcoind_path);
+            let exe_path = bitcoind::exe_path().unwrap();
+
+            let bitcoind = BitcoinD::with_conf(exe_path, &conf).unwrap();
+
+            //--------------------Set Up for Maker & Taker---------------------------------------
+
+            // make rpc_config from bitcoind
+            let url = bitcoind.rpc_url().split_at(7).1.to_string();
+            let auth = Auth::CookieFile(bitcoind.params.cookie_file.clone());
+
+            let rpc_config = RPCConfig {
+                url,
+                auth,
+                network: Network::from_str(rpc_network).unwrap(),
+                wallet_name: wallet_name
+                    .clone()
+                    .unwrap_or_else(|| String::from("WALLET")),
+            };
+
+            let connection_type = read_connection_network_string(network).unwrap();
+
+            // create taker instance
+            let mut taker = Taker::init(
+                data_dir.clone(),
+                wallet_name,
+                Some(rpc_config.clone()),
+                TakerBehavior::Normal,
+                Some(connection_type),
+            )
+            .unwrap();
+
+            // create no of makers based on maker_count
+            let mut makers = Vec::new();
+            let mut base_rpc_port = 3500;
+
+            for i in 0..makers_count {
+                let maker_id = "maker".to_string() + &i.to_string();
+                base_rpc_port += 1;
+
+                let maker = Maker::init(
+                    data_dir.clone(),
+                    Some(maker_id),
+                    Some(rpc_config.clone()),
+                    Some(i),
+                    Some(base_rpc_port),
+                    None,
+                    Some(connection_type),
+                    MakerBehavior::Normal,
+                )
+                .unwrap();
+
+                makers.push(Arc::new(maker));
+            }
```

Comment:
No we can't do Makers this way also. Maker will also be initiated via the std::process::Command call. Once the maker tests are sketched, it will be clearer. Once the Maker, Taker and DNS all are ready with app testing framework, then we will do a whole coinswap round with all 3 of them in a separate test.

Till the just do regular wallet operations with the taker app. 

Like, get a new address, send some funds to the new address, list utxos and balance. Will be good if you can add the `send_to_address` test also. Anything that doesn't require a Maker and DNS should be done here.